### PR TITLE
Update AWS CCM tags

### PR DESCRIPTION
**Pull request by rifelpet**
WIP pending any action from https://github.com/kubernetes/cloud-provider-aws/issues/244

### CODE REVIEW

**Comment by rifelpet** [comment]
https://github.com/kubernetes/cloud-provider-aws/pull/267 has the new location for images (including going back to 1.18)

/test ?

File: upup/models/cloudup/resources/addons/aws-cloud-controller.addons.k8s.io/k8s-1.18.yaml.template [hunk 1]
```diff
@@ -28,7 +28,7 @@ spec:
       serviceAccountName: aws-cloud-controller-manager
       containers:
       - name: aws-cloud-controller-manager
-        image: {{ if .ExternalCloudControllerManager.Image }}{{ .ExternalCloudControllerManager.Image }}{{ else }}gcr.io/k8s-staging-provider-aws/cloud-controller-manager:{{AWSCCMTag}}{{ end }}
+        image: {{ if .ExternalCloudControllerManager.Image }}{{ .ExternalCloudControllerManager.Image }}{{ else }}us.gcr.io/k8s-artifacts-prod/provider-aws/cloud-controller-manager:{{AWSCCMTag}}{{ end }}
```
Comment (olemarkus) [inline upup/models/cloudup/resources/addons/aws-cloud-controller.addons.k8s.io/k8s-1.18.yaml.template:31]: I think we should move this + the tag logic to `pkg/model/components`. Then we can more easily fall back to latest for newer k8s versions.

Also, should the repository for prod be  `k8s.gcr.io/provider-aws/cloud-controller-manager`?

Comment (rifelpet) [reply upup/models/cloudup/resources/addons/aws-cloud-controller.addons.k8s.io/k8s-1.18.yaml.template:31]: I'm trying to move it to pkg/model/components but the cluster spec as seen by `AWSCloudControllerManagerOptionsBuilder`:

https://github.com/kubernetes/kops/blob/8dee92b2f32cbb4e147bbca0f763db01701fb1ae/pkg/model/components/awscloudcontrollermanager.go#L39

shows `cloudControllerManager` as empty, even if the actual cluster spec includes the `cloudProvider` subfield.

---

I think a related symptom is that the integration test input includes the CCM struct and fields:

https://github.com/kubernetes/kops/blob/8dee92b2f32cbb4e147bbca0f763db01701fb1ae/tests/integration/update_cluster/many-addons-ccm/in-v1alpha2.yaml#L19-L20

but the completed cluster spec as stored in the state store has an empty CCM struct:

https://github.com/kubernetes/kops/blob/8dee92b2f32cbb4e147bbca0f763db01701fb1ae/tests/integration/update_cluster/many-addons-ccm/data/aws_s3_bucket_object_cluster-completed.spec_content#L21

I don't know why that is happening but I'm guessing it is related. Any help would be appreciated.

Comment (olemarkus) [reply upup/models/cloudup/resources/addons/aws-cloud-controller.addons.k8s.io/k8s-1.18.yaml.template:31]: If I run `kops get cluster -o yaml --full` I do see:
```
  cloudControllerManager:
    allocateNodeCIDRs: true
    cloudProvider: aws
    clusterCIDR: fd00:10:96::/64
    clusterName: test.kops-dev.srsandbox.io
    configureCloudRoutes: false
```

Comment (olemarkus) [reply upup/models/cloudup/resources/addons/aws-cloud-controller.addons.k8s.io/k8s-1.18.yaml.template:31]: With some rough testing, this only seems to be happening with the terraform tests, but why ... it doesn't make any sense to me.

**Comment by rifelpet** [comment]
I just pushed what ./hack/update-expected.sh is setting for the integration test files. I don't understand why one manifest has `image: null` and another manifest has`image:` removed altogether.

**Comment by rifelpet** [comment]
The external cloud controller manager data seems to be lost during decoding:

https://github.com/kubernetes/kops/blob/d69106e239b8a71048a512b2c58792c416833458/cmd/kops/create.go#L123

The input yaml has the fields but the outputted object does not.

**Comment by rifelpet** [comment]
@olemarkus it turns out the decoding issues were due to the manifests having `provider:` instead of `cloudProvider:`. The test outputs now look as expected and e2e is passing

**Review by olemarkus — APPROVED** [review]

**Pull request merged**